resources/js/alpine/navbarSearchComponent/navbarSearchComponent.ts

```diff
@@ -0,0 +1,149 @@
+import { fetcher } from '@/utils';
+import { computePosition, autoUpdate, ReferenceElement } from '@floating-ui/dom';
+
+type SearchResult = {
+  label: string,
+  mylink: string,
+}
+
+type NavbarSearchComponentType = {
+  showSearchResults:boolean,
+  searchText:string,
+  results: SearchResult[],
+  selectedIndex:number,
+  activeDecendentId: string,
+  init: (formEl: ReferenceElement, ulEl:HTMLElement)=>void,
+  handleClickSearchResult: (label: string) => void,
+  handleKeyDown: (e:KeyboardEvent) => void,
+  handleKeyUp: (e:KeyboardEvent, el:HTMLInputElement) => void,
+}
+
+export function navbarSearchComponent(): NavbarSearchComponentType {
+  return {
+    showSearchResults: false,
+    searchText: '',
+    results: [],
+    selectedIndex: -1,
+
+    init(formEl: ReferenceElement, ulEl: HTMLElement) {
+      if (formEl !== undefined && ulEl !== undefined) {
+        autoUpdate(formEl, ulEl, () => {
+          computePosition(formEl, ulEl, {
+            placement: 'bottom-start'
+          }).then(({
+            x,
+            y
+          }) => {
+            Object.assign(ulEl.style, {
+              left: `${x}px`,
+              top: `${y}px`,
+            });
+          });
+        });
+      }
+    },
+
+    get activeDecendentId() {
+      if (this.selectedIndex !== -1) {
+        return this.results[this.selectedIndex - 2].mylink.slice(1).replace('/', '-');
+      }
+      return '';
+    },
```

Comment:
Is it possible that due to `- 2`, `this.results` could be accessed at a negative index? We may want to have some comments describing why this getter is functioning the way it does; currently my brain is having difficulty reasoning about the logic.

Reply:
I did a small refactor that should make this easier to understand.